### ISSUE TITLE
Compile failure when used on armhf

### ISSUE THREAD

**Issue by jriddell** [KDAB/KDSoap#209]
kio-extras smb now fails to compile but only on 32 bit armhf, it's fine on amd64 and arm64.  I'm told this is an issue from KDSoap.  Any idea why?

https://build.neon.kde.org/job/bionic_unstable_kde_kio-extras_bin_armhf/226/consoleFull

12:05:08 [ 38%] Generating wsdl_wsdd-discovery-1.h
12:05:08 smb/kdsoap-ws-discovery-client/src/CMakeFiles/KDSoapWSDiscoveryClient.dir/build.make:75: recipe for target 'smb/kdsoap-ws-discovery-client/src/wsdl_wsdd-discovery-1.h' failed

generated from this file
https://cgit.kde.org/kio-extras.git/tree/smb/kdsoap-ws-discovery-client/src/wsdiscoveryclient.cpp

**Comment by dfaure-kdab** [comment]
Probably generated from a WSDL file rather than from a CPP file ;)

**Comment by dfaure-kdab** [comment]
The interesting line is the next one:
Illegal instruction (core dumped)

So kdwsdl2cpp crashes. Maybe it's getting compiled for ARM and then run on x86_64? ;)
I would need much more information to treat this as a KDSoap bug, at least a backtrace.
For now it looks to me like a setup issue.

**Comment by caspermeijn** [comment]
> Illegal instruction (core dumped)

Could it be that you are cross compiling? In that case the build process will try to run the kdwsdl2cpp executable of the target architecture. For crosscompiling I have to build a native KDSoap and I have to add something like this too my build:
```
cmake -DKDSOAP_KDWSDL2CPP_COMPILER=/usr/local/bin/kdwsdl2cpp .
```

**Comment by bhush9** [comment]
No this is not cross-compilation setup at all. ARM builds are done natively on native arm machine.

**Comment by jriddell** [comment]
It gets fixed when I use gcc 7.4 to build kdsoap.  Previously it used clang 6.0.

So I guess this can be closed unless you care about clang 6.0.

**Comment by dfaure-kdab** [comment]
We have no way to test clang 6 on armhf, so I'll just close this. To be reopened if someone provides at least a backtrace.

Thanks for the information.
